### PR TITLE
Make pandas stubs partial and use pyright to generate the base set

### PR DESCRIPTION
**Pull request by gramster**
- Deleted all the stub files from /pandas where generating them from the pandas package using pyright instead gives the same or better results.
- Moved what is left to the /partial/pandas directory
- Changed the code to run the tests or generate the stub bundle to take advantage of this; we generate stubs with pyright and then copy over the remaining partial stubs from the partial/pandas directory.
- The remaining pandas stub files have been run through Black for formatting, but otherwise I did not touch the contents. This just makes it easier to compare with the generated stubs (if we run those through Black too) to see what work remains to upstream to pandas.
- Still needed (but to be done separately): move some other partial stubs from pylance to this repo

### CODE REVIEW

File: partial/pandas/core/series.pyi [hunk 1]
```diff
@@ -885,7 +885,7 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
     # just failed to generate these so I couldn't match
     # them up.
     def __add__(self, other: Union[num, _ListLike, Series[S1]]) -> Series[S1]: ...
-    def __and__(self, other: Union[_ListLike, Series[S1], Series[Dtype]]) -> Series[_bool]: ...
+    def __and__(self, other: Union[_ListLike, Series[S1], Series[S1]]) -> Series[_bool]: ...
```
Comment (Dr-Irv) [inline partial/pandas/core/series.pyi:888]: `Series[S1]` is in here twice

**Review by Dr-Irv — COMMENTED** [review]
I did not look at everything - just checked a couple of things that I knew might be issues.

**Comment by gramster** [comment]
> 

One of the things I want to do next is add a bunch more tests. I was planning on taking some from pandas itself but could also look at your snippets.

File: partial/pandas/core/series.pyi [hunk 2]
```diff
@@ -132,7 +132,7 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
     @name.setter
     def name(self, value: Optional[Hashable]) -> None: ...
     @property
-    def values(self) -> ArrayLike: ...
+    def values(self) -> np.ndarray: ...
```
[the remark this answers is not in the document]
Comment (gramster) [reply partial/pandas/core/series.pyi:135]: I don't think we can do that; we can only overload based on the method arguments and there are none. Do you think it is better to leave this as ArrayLike, or should it be a union of the two types you mentioned?

**Comment by gramster** [comment]
Going to try do this using py.typed of partial rather than generating stubs from pyright, so closing this for now and will make a new PR later.

File: partial/pandas/core/series.pyi (hunk 2 above)
[the remark this answers is not in the document]
Comment (Dr-Irv) [reply partial/pandas/core/series.pyi]: I think it should return `ArrayLike` and we change the tests.  `ArrayLike` is defined as the `Union` of `ExtensionArray` and `np.ndarray`

**Pull request closed without merging**